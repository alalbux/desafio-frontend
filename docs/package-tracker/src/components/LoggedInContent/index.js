import React from 'react';
import { Typography, Button } from '@mui/material';
import { useMediaQuery } from '@mui/material';

import LOCALES from '@/locales/navigation';

const { AUTH } = LOCALES;

const LoggedInContent = ({ setIsLogged }) => {
  const isMobile = useMediaQuery('(max-width:768px)');

  return (
    <div style={isMobile ? styles.mobileContainer : styles.desktopContainer}>
      <Typography component="p" sx={styles.loginContent}>
        {AUTH.WELCOME}
      </Typography>
      <Button variant="outlined" onClick={() => setIsLogged(false)}>
        {AUTH.LOGOUT}
      </Button>
    </div>
  );
};

const styles = {
  desktopContainer: {
    display: 'flex',
    flex: 2,
    width: 250,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  mobileContainer: {
    display: 'flex',
    flex: 2,
    width: 240,
    alignItems: 'center',
    marginHorizontal: 10,
    paddingHorizontal: 20,
  },
  loginContent: {
    px: 2,
  },
};

export default LoggedInContent;
