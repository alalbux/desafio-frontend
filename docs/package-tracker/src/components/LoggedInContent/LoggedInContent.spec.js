import React from 'react';
import { render } from '@testing-library/react';
import LoggedInContent from '@/components/LoggedInContent';

describe('LoggedInContent', () => {
  it('matches the snapshot', () => {
    const { container } = render(<LoggedInContent />);
    expect(container).toMatchSnapshot();
  });
});
