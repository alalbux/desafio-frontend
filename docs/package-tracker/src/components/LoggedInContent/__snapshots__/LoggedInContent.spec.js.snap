// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInContent matches the snapshot 1`] = `
<div>
  <div
    style="display: flex; flex: 2; width: 250px; justify-content: space-between; align-items: center;"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-2byhhs-MuiTypography-root"
    >
      Ol√°, User
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Sair
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
